### ISSUE TITLE
Cannot query status of Appveyor projects with periods in the project name

### ISSUE THREAD

**Issue by somewhatabstract** [AlonAm/AnyStatus#14]
### Installed product versions
- Visual Studio: 2015 Enterprise Update 3
- AnyStatus: 1.0
### Description

I have added some status items for my Appveyor projects. They all work fine except the ones that have periods in their project name. For example, `This.Is.My.Project.Name`.
#### Stack

```
[23:31:10.100] Item This.Is.My.Project.Name scheduled to run every 5 minutes.
[23:31:10.100] Updating: This.Is.My.Project.Name.
[23:31:10.554] Failed to update This.Is.My.Project.Name. Exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.AggregateException: One or more errors occurred. ---> System.ArgumentException: Invalid JSON primitive: .
   at System.Web.Script.Serialization.JavaScriptObjectDeserializer.DeserializePrimitiveObject()
   at System.Web.Script.Serialization.JavaScriptObjectDeserializer.DeserializeInternal(Int32 depth)
   at System.Web.Script.Serialization.JavaScriptObjectDeserializer.BasicDeserialize(String input, Int32 depthLimit, JavaScriptSerializer serializer)
   at System.Web.Script.Serialization.JavaScriptSerializer.Deserialize(JavaScriptSerializer serializer, String input, Type type, Int32 depthLimit)
   at System.Web.Script.Serialization.JavaScriptSerializer.Deserialize[T](String input)
   at AnyStatus.Models.AppVeyorBuildHandler.<GetBuildDetailsAsync>d__2.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at AnyStatus.Models.AppVeyorBuildHandler.Handle(AppVeyorBuild item)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at AnyStatus.Infrastructure.ScheduledJob.Execute()
```
### Steps to recreate
1. Have an Appveyor project with periods in it
2. Add a status item to that project
### Current behavior

Indicator doesn't work and there are exceptions in the output log
### Expected behavior

Indicator should work without errors

**Comment by somewhatabstract** [comment]
I suspect this line is to blame:

https://github.com/AlonAm/AnyStatus/blob/master/src/AnyStatus/Features/HealthChecks/AppVeyorBuild.cs#L76

Maybe appveyor needs the periods to be encoded?

**Comment by somewhatabstract** [comment]
Looking at Appveyor, it appears that it replaces periods with `-` and makes the project name lowercase. So `This.Is.My.Project` would have the URL:

`https://ci.appveyor.com/project/AccountName/this-is-my-project`

**Comment by somewhatabstract** [comment]
So, looking at it; there are two ways to "fix" this.

1. Start seeing if there's a way to take a "Project Name" and make it fit the expected slug format
2. Change the field name from "Project Name" to "Project Slug" and have the users type the right thing in

The second option seems like lower overhead and better all round.

**Comment by AlonAm** [comment]
thank you for your effort, i'll review this and get back to ya asap

**Comment by AlonAm** [comment]
fixed by using AppVeyor project slug instead of name as suggested.
Also, existing user settings will upgrade automatically.

**Comment by AlonAm** [comment]
fixed in 0.8.23 now on Visual Studio Gallery
